### PR TITLE
[Reviewer: Ellie] Add log containing pid and errno from findProcess

### PR DESCRIPTION
**Pull request by johadalin**
Added a log to potentially help debugging an issue we saw on the staging system.

The queue manager process had died/been killed, and monit was not restarting it. 
There was still a PID file, with a number in it, but no process with that ID.

Monit logs were repeatedly printing out the error log from https://github.com/Metaswitch/clearwater-monit/blob/master/src/validate.c#L1133 

Hopefully this log will help to debug the issue further if it is seen again.

### CODE REVIEW

**Comment by eleanor-merry** [comment]
As discussed, you're going to split this log into two, adding an error log if the errno is eperm.

**Comment by johadalin** [comment]
Done. 
Also added the braces i was missing so stuff actually works. 

Do i need the braces around the  `if (errno == EPERM) {} ` clause? of should I remove them?

File: src/process/ProcessTree.c [hunk 1]
```diff
@@ -315,8 +315,13 @@ pid_t ProcessTree_findProcess(Service_T s) {
         // Test the cached PID first
         if (s->inf->priv.process.pid > 0) {
                 errno = 0;
-                if (getpgid(s->inf->priv.process.pid) > -1 || errno == EPERM)
+                if (getpgid(s->inf->priv.process.pid) > -1 || errno == EPERM) {
+                        if (errno == EPERM) {
+                                LogError("getpid returned errno == EPERM");
+                        }
+                        DEBUG("getpid returned > -1 for pid %d; errno was %d", s->inf->priv.process.pid, errno);
```
Comment (eleanor-merry) [inline src/process/ProcessTree.c]: *getpgid

**Comment by eleanor-merry** [comment]
You don't *need* them - I'd keep them though

**Pull request merged**